### ISSUE TITLE
Underscores for Folders

### ISSUE THREAD

**Issue by pyronaur** [sass/sass#384]
Ok,
So I'm building my awesome reusable pattern library. And I have about 20 files and growing. So I organize them in folders. (hey even @chriseppstein does it http://d.pr/i/SfqX)

So the idea here is to be able to name a folder with a underscore, instead of naming each and every file in that folder with an underscore. Because let's face it. We all want to compile only one (or maybe for some crazy reason two) CSS File(s). The underscored thing should be the other way around actually :).  
Because that might be a _little bit_ over the top, I would like to think, that having folders with underscores would help everyone.

**Comment by chriseppstein** [comment]
I think this is interesting. Would like to hear other thoughts.

**Comment by Anahkiasen** [comment]
Totally for it. Sass structures are getting more and more complicated and nesting in folders is the way to go (I know Eric Meyer does it) so yeah it makes sense.

**Comment by parkr** [comment]
I think this is a great way to get around the constant underlining of partials. I like this idea.

**Comment by chriseppstein** [comment]
@nex3 thoughts?

**Comment by nex3** [comment]
Seems reasonable.

**Comment by perry** [comment]
Great idea

**Comment by Snugug** [comment]
I do this too. Being able to have something like `/sass/_partials/*` with everything under `_partials` considered a partial would be absolutely awesome for me, and make finding/opening stuff in the terminal much easier.

**Comment by dizzib** [comment]
I currently have my unit tests in a folder named _test containing unit.sass. I'm relying on unit.css to be generated and I'm worried this suggestion would break this.

Why can't partials be declared in code rather than forcing an arbitrary naming convention ? Developers could then use their own naming convention if desired.

**Comment by Anahkiasen** [comment]
No I don't generally like having to specify stuff like that in the code, having a naming convention help see in a quick glance which files are partials and which are not without having to open every one of them.

**Comment by dizzib** [comment]
Fair point. How about having a different file extension for partials, e.g. .sassp or .sasp ? Or perhaps a code directive within a partial to force css generation ?

No matter, at some point I'll probably reorganise my folders to workaround this quirk :) Cheers!

**Comment by zimmi** [comment]
Is this still under consideration / anything I can do to help get this into libsass @nex3?
This would be awesome for static site generators like [zola](https://github.com/getzola/zola/issues/431), allowing to treat entire third party CSS / Sass frameworks as partials (so they aren't built themselves, only when imported).

The alternative is to rename all entry point `.scss` / `.sass` files so they start with an underscore, which is not very good for maintenance.

**Comment by nex3** [comment]
@zimmi If you're interested in contributing this feature, check out [this guide on changing the language](https://github.com/sass/dart-sass/blob/master/CONTRIBUTING.md#changing-the-language).
